core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java

```diff
@@ -53,19 +53,31 @@ public JsonRpc2_0Rx(Web3j web3j, ScheduledExecutorService scheduledExecutorServi
     public Flowable<String> ethBlockHashFlowable(long pollingInterval) {
         return Flowable.create(
                 subscriber -> {
-                    BlockFilter blockFilter = new BlockFilter(web3j, subscriber::onNext);
-                    run(blockFilter, subscriber, pollingInterval);
+                    try {
+                        if(!subscriber.isCancelled()){
```

Comment:
it feels weird that you are checking a property on the subscriber here and then passing the subscriber into the `run(...)` method. Could you not perform this check inside the `run(...)` method to save the duplication of code?